pylint/extensions/check_raise_docs.py

```diff
@@ -0,0 +1,265 @@
+"""Pylint plugin for raised exception documentation checking in Sphinx, Google
+or Numpy style docstrings
+"""
+from __future__ import print_function, absolute_import
+
+import re
+
+import astroid
+
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import node_ignores_exception
+from pylint.extensions.check_docs import space_indentation
+
+
+def _possible_exc_types(node):
+    """
+    Gets all of the possible raised exception types for the given raise node.
+
+    .. note::
+
+        Caught exception types are ignored.
+
+
+    :param node: The raise node to find exception types for.
+
+    :returns: A list of exception types possibly raised by :param:`node`.
+    :rtype: list(str)
+    """
+    excs = []
+    if isinstance(node.exc, astroid.Name):
+        excs = [node.exc.name]
+    elif (isinstance(node.exc, astroid.Call) and
+          isinstance(node.exc.func, astroid.Name)):
+        excs = [node.exc.func.name]
+    elif (node.exc is None and
+          isinstance(node.parent, astroid.ExceptHandler)):
+        handler = node.parent
+        if isinstance(handler.type, astroid.Tuple):
+            excs = [elt.name for elt in handler.type.elts]
+        elif isinstance(handler.type, astroid.Name):
+            excs = [handler.type.name]
+
+    excs = [exc for exc in excs if not node_ignores_exception(node, exc)]
+    return excs
+
+class RaiseDocChecker(BaseChecker):
+    """Checker for raised exception documentation in Sphinx, Google or Numpy
+    style docstrings
+
+    * Check that all explicity raised exceptions in a function are documented
+    in the function docstring. Caught exceptions are ignored.
+
+    Activate this checker by adding the line::
+
+        load-plugins=pylint.extensions.check_raise_docs
+
+    to the ``MASTER`` section of your ``.pylintrc``.
+
+    :param linter: linter object
+    :type linter: :class:`pylint.lint.PyLinter`
+    """
+    __implements__ = IAstroidChecker
+
+    name = 'raise_checks'
+    msgs = {
+        'W9010': ('Raising a "%s" is not documented.',
+                  'missing-raise-doc',
+                  'Please document that this exception type is raised.'),
+    }
+
+    options = (('accept-no-raise-doc',
+                {'default': True, 'type' : 'yn', 'metavar' : '<y or n>',
+                 'help': 'Whether to accept totally missing raises'
+                         'documentation in a docstring of a function.'
+                }),
+              )
+
+    priority = -2
+
+    def __init__(self, linter=None):
+        BaseChecker.__init__(self, linter)
```

Comment:
Changed in upcoming commit. This was taken from `check_docs` so that should be changed at some point also.